course/src/isometry.cc

```diff
@@ -0,0 +1,124 @@
+#include "isometry.h"
+#include <cmath>
+#include "matrix3.h"
+#include "vector3.h"
+
+namespace ekumen {
+namespace math {
+namespace {}  // namespace
+
+Isometry::Isometry(const Vector3& translation, const Matrix3& rotation)
+    : translation_(translation), rotation_(rotation) {}
+
+Isometry::Isometry(const Matrix3& rotation)
+    : translation_(Vector3::kZero), rotation_(rotation) {}
+
+Isometry::Isometry(const Isometry& obj)
+    : Isometry(obj.translation_, obj.rotation_) {}
+
+Isometry::Isometry(Isometry&& obj)
+    : translation_(std::move(obj.translation_)),
+      rotation_(std::move(obj.rotation_)) {}
+
+Isometry& Isometry::operator=(const Isometry& obj) {
+  translation_ = obj.translation_;
+  rotation_ = obj.rotation_;
+  return *this;
+}
+Isometry& Isometry::operator=(Isometry&& obj) {
+  // The rvalue reference shouldn't be the same as this.
```

Comment:
Usually you deal with this by just not doing anything. Don't throw, just exit the function. Let moving into itself be a no-operation.

Reply:
Ack. Now it's a no-op. removed tests to assert that it throws exception.